mmv1/provider/template_data.go

```diff
@@ -203,7 +203,14 @@ func (td *TemplateData) GenerateTGCIamResourceFile(filePath string, resource api
 func (td *TemplateData) GenerateFile(filePath, templatePath string, input any, goFormat bool, templates ...string) {
 	templateFileName := filepath.Base(templatePath)
 
-	tmpl, err := template.New(templateFileName).Funcs(google.TemplateFunctions).ParseFiles(templates...)
+	funcMap := template.FuncMap{
+		"TemplatePath": func() string { return templatePath },
+	}
+	for k, v := range google.TemplateFunctions {
```

Comment:
Unfortunate we can't get TemplatePath in the original TemplateFunctions map, but I can't think of a better way to pass the path outside of the `input` object

Reply:
yeah I thought we could like have a function to return different mapping of templates but that seems excessive